archunit/src/test/java/com/tngtech/archunit/lang/syntax/elements/MembersShouldTest.java

```diff
@@ -653,6 +660,65 @@ public void haveNameNotEndingWith(ArchRule rule, String suffix) {
                         locationPattern(SimpleFieldAndMethod.class)));
     }
 
+    @DataProvider
+    public static Object[][] containNumberOfElements_rules_three_members() {
+        return $$(
+                $(equalTo(999)),
+                $(lessThan(0)),
+                $(lessThan(1)),
+                $(lessThan(2)),
+                $(lessThan(3)),
+                $(greaterThan(3)),
+                $(greaterThan(4)),
+                $(greaterThan(999)),
+                $(lessThanOrEqualTo(0)),
+                $(lessThanOrEqualTo(1)),
+                $(greaterThanOrEqualTo(4)),
+                $(greaterThanOrEqualTo(999)));
+    }
+
+    @Test
+    @UseDataProvider("containNumberOfElements_rules_three_members")
+    public void containNumberOfElements_parameter_and_method_with_same_name(DescribedPredicate<Integer> predicate) {
+        EvaluationResult result = members().should().containNumberOfElements(predicate).evaluate(importClasses(SimpleFieldAndMethod.class));
+
+        assertThat(singleLineFailureReportOf(result))
+                .contains("contain number of elements " + predicate.getDescription())
+                .contains("there is/are 3 element(s) in ["
+                        + "com.tngtech.archunit.lang.syntax.elements.testclasses.SimpleFieldAndMethod.<init>(), "
+                        + "com.tngtech.archunit.lang.syntax.elements.testclasses.SimpleFieldAndMethod.violated, "
+                        + "com.tngtech.archunit.lang.syntax.elements.testclasses.SimpleFieldAndMethod.violated()]");
+    }
+
+    @DataProvider
+    public static Object[][] containNumberOfElements_rules_two_members() {
+        return $$(
+                $(equalTo(999)),
+                $(lessThan(0)),
+                $(lessThan(1)),
+                $(lessThan(2)),
+                $(greaterThan(2)),
+                $(greaterThan(3)),
+                $(greaterThan(999)),
+                $(lessThanOrEqualTo(0)),
+                $(lessThanOrEqualTo(1)),
+                $(greaterThanOrEqualTo(3)),
+                $(greaterThanOrEqualTo(999)));
+    }
+
+    @Test
+    @UseDataProvider("containNumberOfElements_rules_two_members")
+    public void containNumberOfElements_two_methods_with_same_name(DescribedPredicate<Integer> predicate) {
```

Comment:
I don't exactly understand why we need both these tests? One test that has a constructor, field and method seems good enough? :thinking: What's the additional test coverage we're getting here? The "full name" is not the same, because the full name includes the parameter names :man_shrugging:

Reply:
I added this test, because I was not sure how the method `getFullName` works, and thus I wanted to test, that for two methods with the same name, but different parameters, the "full name" is different. But I agree, that we can probably remove this test, since we do not want to test in this test that the method `getFullName` works as expected.